docs/APISPEC.md

```diff
@@ -2,221 +2,1225 @@
 ```
+<details>
+  <summary>Error examples:</summary>
+  
+| Error | Error code | Body | 
+| --- | --- | --- | 
+| Friendcode is already used for adding a friend | 403 Forbidden | { "error": "Already friends"} |
+| Friendcode from body request is not found | 404 Not Found | { "error": "User not found"} |
+| Friendcode matches with friendcode of authorized user himself the n 403 Forbidden | 403 Forbidden | { "error": "You cannot add yourself"} |
+</details> 
 
-### POST /users/changepassword
+#### <a name="list_friends"></a>  [2. GET friends/list](#friends)
 
-Changes the users password
+Gets a list of added user friends
+    
+<details>
+  <summary>Header params:</summary>
+  
+| Name |  Value | 
+| --- | --- | 
+| Authorization | Bearer `<token>` |
+</details> 
 
-Accepts:
+**Sample request**
+
+```curl
+curl --request GET 'https://testaustime.fi/api/friends/list' \
+--header 'Authorization: Bearer <token>'
+```
+    
+**Sample response** 
+```JSON
+[
+    {
+        "username": "username",
+        "coding_time": {
+            "all_time": 0,
+            "past_month": 0,
+            "past_week": 0
+        }
+    }
+]
 ```
+
+<details>
+  <summary>Response definitions:</summary>
+  
+| Response Item | Type | Description | 
+| --- | --- | --- | 
+| username | string | Friend's username |
+| coding_time | Object | Coding friend's time by total, past month and past week | 
+| all_time | int | Total duration of user code sessions in seconds |
+| past_month | int| Total duration of user code sessions in seconds for past month |
+| past_week | int| Total duration of user code sessions in seconds for past week |
+</details> 
+
+#### <a name="regenerate_fc"></a>  [3. POST /friends/regenerate](#friends)
+
+Regenerates the authorized user's friend code
+
+<details>
+  <summary>Header params:</summary>
+  
+| Name |  Value | 
+| --- | --- | 
+| Authorization | Bearer `<token>` |
+</details> 
+
+**Sample request**
+```curl
+curl --request POST 'https://testaustime.fi/api/friends/regenerate' \
+--header 'Authorization: Bearer <token>'
+```
+    
+**Sample response**
+```JSON
 {
-    "old": string,
-    "new": string
+    "friend_code": "friend_code"
 }
 ```
 
-Required headers:
+<details>
+  <summary>Response definitions:</summary>
+  
+| Response Item | Type | Description | 
+| --- | --- | --- | 
+| friend_code | string| New friend code. Using for the all next create friends paire operations |
+</details> 
+
+#### <a name="remove_friend"></a>  [4. DELETE /friends/remove](#friends)
+
+Removes another user from your friend list
+    
+<details>
+  <summary>Header params:</summary>
+  
+| Name |  Value | 
+| --- | --- | 
+| Authorization | Bearer `<token>` |
+</details> 
+
+<details>
+  <summary>Body:</summary>
+  
+| Param | Type | Description | 
+| --- | --- | --- | 
+| raw text | string | Should contain username without any prefixes |
+</details> 
+
+**Sample request** 
+```curl 
+--location --request DELETE 'https://testaustime.fi/api/activity/delete' \
```

Comment:
Why does the example not include the `curl` command itself.
The endpoint is also incorrect. Shouldn't it be `/friends/remove`?
Additionally change path to https://api.testaustime.fi